### PR TITLE
Galaxies

### PR DESCRIPTION
**Pull request by nicl**
*To test locally, add `GALAXIES_BUCKET_NAME` to your `.env` (ask me for the bucket name). And make sure you have Janus creds.*

Adds the following endpoints:

```
GET /teams
GET /teams/:id
GET /people
```

This provides us with Galaxies team/people info but collected from S3 - as Galaxies is not yet machine-accessible.

The Galaxies data has been augmented with some Github mappings:

* the github username for people
* the primary github team for teams

These were produced with a quick script and also some manual work. Ultimately we should aim to integrate this data directly into Galaxies (or, more specifically, the spreadsheet it uses for its data).

Note, the S3 data contains errors but is enough to get started. Separate to this PR, we will work to integrate this data directly into Galaxies and replace our S3 Galaxies API implementation with actual API calls.

--

TODO: 

* ~add CDK permissions to access the tmp Galaxies bucket (this PR)~ (DONE)
* write a script to copy the spreadsheet mappings into S3 as JSON (separate)
* integrate mappings directly into Galaxies (separate)

### CODE REVIEW

File: packages/services-api/src/camelcase.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
+// Inlined with simplification and types added from
```
[the remark this answers is not in the document]
Comment (akash1810) [reply packages/services-api/src/camelcase.ts:1]: ESM is wonderful. We could use [v6.3.0](https://github.com/sindresorhus/camelcase/releases).

File: packages/services-api/src/galaxies.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+			this.client,
+			this.bucket,
+			'teams-mapping.json',
+		);
+
+		const teamsMapping = mappingResp.payload;
+
+		// Add the (for now) missing fields.
+		return Object.values(teams).map((team) => {
+			const teamId = camelCase(team.teamName);
```
[the remark this answers is not in the document]
Comment (nicl) [reply packages/services-api/src/galaxies.ts:95]: Passing query/path params that include spaces is always a bit gnarly. camelCase is what Galaxies uses for lookup too (I hadn't realised this when I wrote this) so it is consistent with them at least.

File: packages/services-api/src/galaxies.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	teams: string[];
+	githubUsername: string; // Note, this doesn't (yet) exist in actual Galaxies.
+}
+
+interface GalaxyTeam {
+	teamName: string;
+	teamDescription: string;
+	teamContactEmail: string;
+	teamGoogleChatSpaceKey: string;
+	id: string; // Note, this doesn't (yet) exist in actual Galaxies.
```
[the remark this answers is not in the document]
Comment (nicl) [reply packages/services-api/src/galaxies.ts:20]: Name is like 'Editorial Tools` and id is `editorialTools`.

File: packages/services-api/src/galaxies.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicl) [reply packages/services-api/src/galaxies.ts:95]: In fact, we can use the keys directly and drop this library altogether! So will do that.

**Comment by kenoir** [comment]
Not required here, but perhaps we could push the work done via the script to associate GitHub username and google email / galaxies identity into a lambda?

File: packages/services-api/package.json [hunk 4]
```diff
@@ -29,6 +29,7 @@
     "typescript": "^4.8.4"
   },
   "dependencies": {
+    "@aws-sdk/credential-providers": "^3.212.0",
```
[the remark this answers is not in the document]
Comment (nicl) [reply packages/services-api/package.json:32]: Ah the v2 dep is an error - have removed.

**Review by kenoir — APPROVED** [review]
Maybe also commit the script that generated the files in this repo in another PR?

File: packages/cdk/lib/__snapshots__/services-api.test.ts.snap [hunk 5]
```diff
@@ -65,6 +65,11 @@ Object {
       "Description": "Virtual Private Cloud to run EC2 instances within. Should NOT be the account default VPC.",
       "Type": "AWS::SSM::Parameter::Value<AWS::EC2::VPC::Id>",
     },
+    "galaxiesbucketname": Object {
+      "Default": "/INFRA/deploy/services-api/galaxiesBucketName",
```
Comment (akash1810) [inline packages/cdk/lib/__snapshots__/services-api.test.ts.snap:69]: minor: Should we have a consistent casing here? We've kebab and title.

**Review by akash1810 — APPROVED** [review]
LGTM

**Pull request merged**